python/cudf/cudf/core/column/decimal.py

```diff
@@ -224,6 +224,13 @@ def fillna(
 
         Returns a copy with null filled.
         """
+        if isinstance(value, (int, Decimal)):
+            value = pa.scalar(
+                value,
+                type=pa.decimal128(self.dtype.precision, self.dtype.scale),
```

Comment:
I think the following should work:

```suggestion
                type=sellf.dtype.to_arrow()
```